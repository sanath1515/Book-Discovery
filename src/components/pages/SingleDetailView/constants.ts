import { bookProp } from "../../../data/type";

type Image = {
    image: string;
  };
  
const image_1: Image = {
    image:
      "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/843192ED-FF34-4571-B8AA-653DDE8C1434.png",
  };
  const image_2: Image = {
    image:
      "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/03991C69-3988-432F-AD72-535D7FCFFC54.png",
  };
  const image_3: Image = {
    image:
      "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/1C868AAA-E223-420A-95EE-01A58145DC1A.png",
  };
export const authorBooks: Image[] = [image_1, image_2, image_3];
  
export const batchmateBooks = [
    {
      image:
        "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/A78B8C90-C12E-45B1-9523-2C24CCA913ED.png",
    },
    {
      image:
        "https://images-eu.ssl-images-amazon.com/images/I/61JPFJpEZxL._SY445_SX342_QL70_ML2_.jpg",
    },
    {
      image: "https://images-na.ssl-images-amazon.com/images/I/71u9RfUhiPL.jpg",
    },
    {
      image: "https://images-na.ssl-images-amazon.com/images/I/81OrIk+OffL.jpg",
    },
  ];

  export const reviewCardData=[
      {
        image:"https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/8D1CA0C4-4BED-4616-A2B3-28DC74375748.png",
        name:"Keerthi kumar"
      },
      {
        image:"https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/8911A6F3-BE2F-4473-B410-40CA75BF34C6.png",
        name:"Mounavi Ramesh"
      },
      {
        image:"https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/302EA847-4DEE-4DE2-B8A0-DCC9CE94AA2D.png",
        name:"Pranitha"
      }
  ]

export const defaultData:bookProp=
  {
    "id": "1",
    "title": "Crystal chemistry",
    "author": "Carlos Gonzalez",
    "category": "Chemistry",
    "pagesLeft": 100,
    "totalPages": 200,
    "rating": 5,
    "bookmark": false,
    "addToLibrary": false,
    "language": "English",
    "numberOfRatings": 1200,
    "bookImage": "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/E3EEB4FA-4DF1-491D-8A51-DC1C1921FB60.png",
    "authorImage": "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/8911A6F3-BE2F-4473-B410-40CA75BF34C6.png",
    "releaseDate": "12 sep 1998",
    "pagesRead": "100",
    "desc": " Through a series of recent breakthroughs, deep learning has boosted the entire field of machine learning.Now, even programmers who know close to nothing about this technology can use simple, efficient tools to implement programs capable of learning from data.This practical book shows you how.By using concrete examples, minimal theory, and two production-ready Python frameworks—Scikit-Learn and TensorFlow—author Aurélien Géron helps you gain an intuitive understanding of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks. With exercises in each of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks."
  }
