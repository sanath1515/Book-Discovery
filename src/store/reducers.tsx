export type bookState = {
    searchText:string,
    category:string,
    singleViewId:string
}

export const initialBookState = {
    searchText:"",
    category:"",
    singleViewId:"1"
}

type Action = {type:string,payload:string}

export const booksReducer = (state:bookState = initialBookState,action:Action)=>
{
    switch(action.type)
    {
        case "SET_CATEGORY":{
            return {...state,category:action.payload};
        }
        case "SET_SEARCH_TEXT":{
            return {...state,searchText:action.payload};
        }
        case "SET_SINGLE_VIEW_ID":{
            return {...state,singleViewId:action.payload};
        }
        default:
            return state;
    }
}