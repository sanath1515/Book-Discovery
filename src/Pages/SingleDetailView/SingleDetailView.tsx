import React, { useEffect, useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import BookDetail from "../../organisms/BookDetail/BookDetail";
import Header from "../../organisms/Header/Header";
import Batchmates from "../../organisms/Batchmates";
import { Grid } from "@material-ui/core";
import AuthorItem from "../../organisms/Author/Author";
import StudentReview from "../../organisms/ReviewCardMock";
import ReviewCard from "../../organisms/ReviewCard";
import RecommendationContainer from "../../organisms/RecommendationContainer/RecommendationContainer";
import theme from "../../../theme/theme";
import {
  batchmateBooks,
  authorBooks,
  reviewCardData,
  defaultData,
} from "./constants";
import { useSelector } from "react-redux";
import { bookState } from "../../../store/reducers";
import { bookProp } from "../../../data/type";

const useStyles = makeStyles({
  main: {
    margin: "auto",
    width: "100%",
    maxWidth: "1440px",
  },
  submain: {
    margin: "auto",
    width: "100%",
    maxWidth: "1110px",
    justifyContent: "space-between",
  },
  container: {
    marginLeft: theme.spacing(7),
  },
  bookDetail: {
    margin: theme.spacing(7, 0, 7, 0),
  },
  batchmates: {
    margin: theme.spacing(7, 0, 3, 0),
  },
  wrap: {
    flexWrap: "nowrap",
    justifyContent: "space-between",
  },
});

export type SingleDetailViewProps = {
  id?: string;
};
const SingleDetailView: React.FC<SingleDetailViewProps> = (props) => {
  const classes = useStyles();

  const singleViewIdStore = useSelector<bookState, bookState["singleViewId"]>(
    (state) => state.singleViewId
  );

  const [allBooks, setAllBooks] = useState<bookProp[]>([]);
  const [book, setBook] = useState<bookProp>(defaultData);

  useEffect(() => {
    getBook(singleViewIdStore);
    getAllBooks();
  }, [singleViewIdStore]);

  const getBook = async (id: string) => {
    await fetch("http://localhost:3000/books/" + id)
      .then((res) => {
        return res.json();
      })
      .then((data) => {
        setBook(data);
      });
  };

  const getAllBooks = async () => {
    await fetch("http://localhost:3000/books")
      .then((res) => {
        return res.json();
      })
      .then((data) => {
        setAllBooks(data);
      });
  };

  const reviewCards = reviewCardData.map((element) => {
    return (
      <ReviewCard image={element.image} name={element.name} children={[]} />
    );
  });
  return (
    <div className={classes.main}>
      <Header />
      <div className={classes.submain}>
        <Grid direction="row" container className={classes.wrap}>
          <Grid item className={classes.bookDetail}>
            <BookDetail
              authorName={book!.author!}
              desc={book!.desc!}
              fieldName={book!.category!}
              image={book!.bookImage!}
              language={book!.language!}
              releaseDate={book!.releaseDate!}
              title={book!.title!} pagesRead={book!.pagesRead!} totalPages={book!.totalPages!}            />
          </Grid>
          <Grid item className={classes.batchmates}>
            <Batchmates booksSrc={batchmateBooks} />
          </Grid>
        </Grid>
        <AuthorItem
          authorName={book!.author!}
          authorImage={book!.authorImage!}
          booksSrc={authorBooks}
        />
        <StudentReview />
        <Grid>{reviewCards}</Grid>
        <RecommendationContainer
          mainHeading="Recommendations based on your search"
          data={allBooks.slice(0, 7)}
          className={classes.container}
        />
      </div>
    </div>
  );
};

export default SingleDetailView;
