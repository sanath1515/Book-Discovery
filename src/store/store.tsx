import { createStore } from 'redux';
import { booksReducer } from './reducers';

const store = createStore(booksReducer);

export default store;