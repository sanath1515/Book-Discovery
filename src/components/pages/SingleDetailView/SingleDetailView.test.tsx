import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import store from "../../../store/store";
import SingleDetailView from "./SingleDetailView";

jest.mock('../ApiProvider', () => ({
  ...(jest.requireActual('../ApiProvider')),
  getBook: jest.fn(),
  getAllBooks: jest.fn().mockImplementation(() => ({ type: '' })),
}));

describe("Single Detail Page Test", () => {
  
  const page = ()=>{render(
    <Router>
      <Provider store={store}>
        <SingleDetailView />
      </Provider>
    </Router>
  );}

  it("Match Snapshot", () => {
    expect(page).toMatchSnapshot();
  });

  it("To be Required",()=>{
    const container =()=>render(
      <Router>
        <Provider store={store}>
          <SingleDetailView />
        </Provider>
      </Router>
    );
    expect(container).toBeRequired;
  })
});
