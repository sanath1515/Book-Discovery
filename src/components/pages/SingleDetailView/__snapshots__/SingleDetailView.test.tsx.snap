// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Detail Page Test Match Snapshot 1`] = `[Function]`;
